test/extensions/stats_sinks/hystrix/hystrix_test.cc

```diff
@@ -0,0 +1,433 @@
+                                    const uint64_t timeout_step2) {
+
+    // Add new cluster.
+    addSecondClusterHelper(buffer);
+
+    // Generate data to both clusters.
+    for (uint64_t i = 0; i < (window_size_ + 1); i++) {
+      buffer.drain(buffer.length());
+      cluster1_.setCounterReturnValues(i, success_step, error_step, 0, 0, 0, timeout_step, 0, 0);
+      cluster2_.setCounterReturnValues(i, success_step2, error_step2, 0, 0, 0, timeout_step2, 0, 0);
+      sink_->flush(source_);
+    }
+
+    return buildClusterMap(TestUtility::bufferToString(buffer));
+  }
+
+  void removeSecondClusterHelper(Buffer::OwnedImpl& buffer) {
+    buffer.drain(buffer.length());
+    removeClusterFromMap(cluster2_name_);
+    sink_->flush(source_);
+  }
+
+  void validateResults(const std::string& data_message, uint64_t success_step, uint64_t error_step,
+                       uint64_t timeout_step, uint64_t timeout_retry_step, uint64_t rejected_step,
+                       uint64_t window_size) {
+    EXPECT_EQ(getStreamField(data_message, "rollingCountSemaphoreRejected"),
+              std::to_string(window_size * rejected_step));
+    EXPECT_EQ(getStreamField(data_message, "rollingCountSuccess"),
+              std::to_string(window_size * success_step));
+    EXPECT_EQ(getStreamField(data_message, "rollingCountTimeout"),
+              std::to_string(window_size * (timeout_step + timeout_retry_step)));
+    EXPECT_EQ(getStreamField(data_message, "errorCount"),
+              std::to_string(window_size *
+                             (error_step - timeout_step))); // Note that on regular operation,
+    // 5xx and timeout are raised together, so timeouts are reduced from 5xx count
+    uint64_t total = error_step + success_step + rejected_step + timeout_retry_step;
+    EXPECT_EQ(getStreamField(data_message, "requestCount"), std::to_string(window_size * total));
+    if (total != 0) {
+      EXPECT_EQ(
+          getStreamField(data_message, "errorPercentage"),
+          std::to_string(static_cast<uint64_t>(
+              100 * (static_cast<double>(total - success_step) / static_cast<double>(total)))));
+    } else {
+      EXPECT_EQ(getStreamField(data_message, "errorPercentage"), "0");
+    }
+  }
+
+  std::unordered_map<std::string, std::string> buildClusterMap(absl::string_view data_message) {
+    std::unordered_map<std::string, std::string> cluster_message_map;
+    std::vector<std::string> messages =
+        absl::StrSplit(data_message, "data:", absl::SkipWhitespace());
+    for (auto message : messages) {
+      if (absl::StrContains(getStreamField(message, "type"), "HystrixCommand")) {
+        std::string cluster_name(getStringStreamField(message, "name"));
+        cluster_message_map[cluster_name] = message;
+      }
+    }
+    return cluster_message_map;
+  }
+
+  TestRandomGenerator rand_;
+  uint64_t window_size_ = rand_.random() % 10 + 5; // Arbitrary reasonable number.
```

Comment:
nice!  I didn't realize this but, but I just poked around, and this both echoes the seed, and provides a way for you to re-run with the same seed.